### PR TITLE
Fix back button arrow won't back when viewing attachment image and additional fixes on modals

### PR DESCRIPTION
**Pull request by tsa321**
Fix back button arrow won't back when viewing attachment image and fix failure to open search page with shortcut when viewing some modals.

### Fixed Issues

$ https://github.com/Expensify/App/issues/42368
PROPOSAL: https://github.com/Expensify/App/issues/42368#issuecomment-2119658251

### Tests

1. Open a report with 3 or more attachment images
2. View first attachment images
3. Press browser's back button (for non browser test, press back button on phone or tap on close modal button)
4. Verify the image modal is closed and report screen displayed properly
5. Open Fab menu, Three dots menu, Open search page, Request money, Profile page, preferences page, workspace page, Right click report actions, right click on LHN and other pages or popover
6. The pages/popover should open normally
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open a report with 3 or more attachment images
2. View first attachment images
3. Press browser's back button (for non browser test, press back button on phone or tap on close modal button)
4. Verify the image modal is closed and report screen displayed properly
5. Open Fab menu, Three dots menu, Open search page, Request money, Profile page, preferences page, workspace page, Right click report actions, right click on LHN and other pages or popover
6. The pages/popover should open normally
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/114975543/a3d66a38-f0e7-4881-a99a-5c6530eeaf9f

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/114975543/25104bc0-67a6-4423-a914-13e68f885c9f

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/114975543/e0a35aeb-a5b5-4fa0-a544-c53788b3bde3

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/114975543/ae5cf69e-e8cb-495a-a763-d62b437b3b85

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/114975543/0b8e4239-a7ef-4f77-8d63-047db1980c37

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/114975543/77465aa3-9663-4fb6-8d80-1473d7eda476

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@tsa321 Please also test other modals to ensure our change don't cause any regression

**Comment by tsa321** [comment]
@DylanDylann yes, I have tested it on other modals too.

**Comment by DylanDylann** [comment]
It will take me a bit of time to ensure this change is safe

**Comment by DylanDylann** [comment]
I am about to request that you merge the latest main 😄 . Thanks for always keeping eye on this PR

File: src/components/Modal/BaseModal.tsx [hunk 1]
```diff
@@ -141,6 +141,7 @@ function BaseModal(
     };
 
     const handleDismissModal = () => {
+        hideModal();
```
Comment (DylanDylann) [inline src/components/Modal/BaseModal.tsx:144]: @tsa321 I can't reproduce https://github.com/Expensify/App/issues/27237 on the latest main. Should we revert this change?

Comment (tsa321) [reply src/components/Modal/BaseModal.tsx:144]: Yes, I will revert BaseModal file

Comment (tsa321) [reply src/components/Modal/BaseModal.tsx:144]: @DylanDylann I cannot reproduce it in staging, but I can reproduce it in latest main. 
It is even more broken in latest main, even opening receipt modal -> cmd +`k` won't open search page. Could you check it?

Also my code depends on the `hideModal` function inside BaseModal, but when closing  receipt modal the `hideModal` function is not executed. So I make that modification in BaseModal. 
Could you check it thank you...

Comment (DylanDylann) [reply src/components/Modal/BaseModal.tsx:144]: @tsa321 Try again and still can't reproduce on the latest main

Comment (DylanDylann) [reply src/components/Modal/BaseModal.tsx:144]: Anyway, I think it is out of the scope of this PR, we should revert that to avoid delay processing

Comment (tsa321) [reply src/components/Modal/BaseModal.tsx:144]: @DylanDylann alright. I will make minimal changes. 
If I make modification in the beforeRemove I need the changes in baseModal, so the composer will re-focus after user close receipt modal.
But I will move the condition in shouldAutoFocus of ComposerWithSuggestions and revert  the changes in BaseModal and AuthScreen. So the changes will be minimal.

**Comment by tsa321** [comment]
@DylanDylann I have reverted the changes in basemodal and authscreen. I move the condition check inside composerwithsuggestion. The changes should be minimal now.

**Review by DylanDylann — APPROVED** [review]

**Review by dangrous — APPROVED** [review]
awesome! this ended up cleaner than I expected

**Comment by luacmartins** [comment]
I think this PR caused [this regression](https://github.com/Expensify/App/issues/43222), more specifically commit [29992cf894d0ebcdcb2e20311435638ee1dedf58](https://github.com/Expensify/App/commit/29992cf894d0ebcdcb2e20311435638ee1dedf58)

**Comment by luacmartins** [comment]
@tsa321 I'll demote that blocker since it's a small issue, but can you take a look and work on a fix please?

**Comment by luacmartins** [comment]
FYI although there's no staging deploy message here, this PR is on the staging checklist https://github.com/Expensify/App/issues/43168

**Pull request merged**